### PR TITLE
Gitops e2e for vSphere

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*
This PR adds vSphere E2E tests for the API with GitOps

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/workload_api_flux_test.go [hunk 1]
```diff
@@ -202,3 +249,95 @@ func TestDockerUpgradeWorkloadClusterScaleAddRemoveWorkerNodeGroupsGitHubFluxAPI
...
+			),
+			api.VSphereToConfigFiller(api.WithDisableCSI(true)),
+			vsphere.WithUbuntu124(),
+		)
+		wc.ValidateClusterState()
+		test.DeleteWorkloadClusterFromGit(wc)
+		wc.ValidateClusterDelete()
+	})
+	test.ManagementCluster.StopIfFailed()
+	test.DeleteManagementCluster()
```
Comment (cxbrowne1207) [inline test/e2e/workload_api_flux_test.go:342]: You could probably use `runWorkloadClusterUpgradeFlowAPIWithFlux` here as well?

**Review by cxbrowne1207 — COMMENTED** [review]
Looks good. Left a comment

File: test/e2e/workload_api_flux_test.go (hunk 1 above)
Comment (tatlat) [reply test/e2e/workload_api_flux_test.go:342]: i forgot to delete the csi resources in this e2e, which was the main reason i didn't use runWorkloadClusterUpgradeFlowAPIWithFlux

**Review by cxbrowne1207 — APPROVED** [review]

**Pull request closed without merging**